lib/format.py

```diff
@@ -24,18 +26,44 @@ class Format:
     name = 'Base Format'
     filefilter = FileFilter('Text Files', 'txt')
 
-    def __init__(self, unitsMgr: adsk.core.UnitsManager, docname: str, units=None):
+    def __init__(self, unitsMgr: adsk.core.UnitsManager, docname: str, units=None, fractions=False):
         self.unitsMgr = unitsMgr
         self.docname = docname
         self.units = units if units else unitsMgr.defaultLengthUnits
+        self.fractions = fractions
 
     @property
     def filename(self):
         name = self.docname.lower().replace(' ', '_')
         return f'{name}.{self.filefilter.ext}'
 
     def format_value(self, value, showunits=False):
-        return self.unitsMgr.formatInternalValue(value, self.units, showunits)
+        result = self.unitsMgr.formatInternalValue(value, self.units, showunits)
+
+        if not self.fractions:
+          return result
+
+        match = re.match("(\d+)(\.\d+)?( (?:in|cm|mm))?", result)
```

Comment:
I think it should be possible to avoid using a regular expression here and to compute the value more directly by taking advantage of some additional methods on the unit manager:

1. If not using fractions, call `formatInternalValue` and return the result
2. Otherwise, call [`convert`](https://help.autodesk.com/view/fusion360/ENU/?guid=GUID-20475e06-1d24-4586-bc0a-96f21d00fac9) with [`internalUnits`](https://help.autodesk.com/view/fusion360/ENU/?guid=GUID-7c2df7ad-aeea-49c1-96b5-71382f8d520f) as the input unit and `self.units` as the output
3. Use [`math.modf`](https://docs.python.org/3/library/math.html#math.modf) to separate the float value in to the integer and fractional parts
4. Do the fractional formatting
5. Call [`formatUnits`](https://help.autodesk.com/view/fusion360/ENU/?guid=GUID-f1446930-05a7-4b28-935f-b708fe82b1a8) to generate the correct unit string from `self.units`

Reply:
Thank you for the API links! 

Hmm.

I think `math.modf` can't be used here, because it will convert the incoming value to a float, giving unexpected results for some values, and that it may be better to split the value as a string.

Because of that, I think the (less regex-ey) may not be as much of an improvement:

```python
converted_value = self.unitsMgr.convert(value, self.unitsMgr.internalUnits, self.units)

split_value = str(converted_value).split('.')
integerValue = split_value[0]
if len(split_value) > 1:
  decimalValue = split_value[1]

# further string operations on integer/decimal parts
```

Reply:
> Call [formatUnits](https://help.autodesk.com/view/fusion360/ENU/?guid=GUID-f1446930-05a7-4b28-935f-b708fe82b1a8) to generate the correct unit string from self.unit

Is that necessary? It seemed like the internal units are already being passed around as in their abbreviated form, which `formatUnits` converts to.

Reply:
Unlike the formatting methods, the `convert` method returns a float value, so there should be no additional conversion. That means the main difference compared to using the string is that it doesn't respect the user's setting for decimal precision. I think having the full precision for fraction formatting is probably a net positive, since I think the default 3-decimal precision would end up rounding 1/16ths and smaller such that they didn't map to power-of-2 denominators.

Regarding `formatUnits`: looking at the code again, in this particular case, you're right that it isn't needed. The `showunits` parameter is never used, so the units are never included in these values in practice. In the general case, the Fusion preferences have an option that controls whether to use abbreviated or full unit names. If the user selects full unit names, the formatting functions expand the abbreviations to the full names... but I forgot to call that in the places where units are currently used, so I think they always appear as the abbreviations.

My goal was to respect the global preferences for formatting by default, although I guess with your addition of overrides, there might be an argument for cutlist export always doing its own thing.

By the way, if I switch the "Foot and inch display format" setting to "Fractional", the current version of the addin will use fractional values in the export. That also means that whether the output of `formatInternalValue` is a fraction or a decimal depends on user settings.

<details>
<summary>Available formatting settings for reference</summary>

![Screenshot 2024-06-30 203504](https://github.com/bluekeyes/Fusion360-ExportCutlist/assets/1745813/cfd6304c-7303-4c05-bee5-885d1257e341)

</details>